### PR TITLE
proxy: Refactor `convert` from crate to module

### PR DESCRIPTION
**Pull request by olix0r**
In e2093e3, we created a `convert` crate when refactoring the proxy's
gRPC bindings into a dedicated crate.

It's not really necessary to handle `convert` as a crate, given that it
holds a single 39-line file that's mostly comments. It's possible to
"vendor" this file in the proxy and controller-grpc crates because
neither crate requires that the other use the same type.

### CODE REVIEW

**Comment by seanmonstar** [comment]
Does doing this provide some benefit? If so, it'd be useful to have that here in the comment history for when we forget.

**Comment by briansmith** [comment]
What are we trying to accomplish with this? I agree that it is kind of silly to have a separate crate for this but it also seems bad to duplicate it.

I don't believe having it as a separate trait is really conferring any benefit to us now. Instead, I suggest we just don't use the `TryFrom` trait in the GRPC bindings module; instead we can just implement `try_from` as a regular method on the types that implement it.

(Many of the times we've tried to use `TryFrom` in the proxy, the reviewer has asked for the function to be given a more descriptive name. I suspect that some of our current uses of `TryFrom` are also actually more confusing than if we'd just used an inherent method, perhaps with a better name.)

**Comment by olix0r** [comment]
@hawkw that was the motivation for splitting controller-grpc into its own crate. this one trait isn't going to make a dent in our compile times.

File: proxy/controller-grpc/src/lib.rs [hunk 1]
```diff
@@ -193,9 +191,8 @@ impl<'a> From<&'a ::std::net::SocketAddr> for common::TcpAddress {
 
 // ===== impl common::scheme::Type =====
 
-impl<'a> TryInto<String> for &'a common::scheme::Type {
-    type Err = InvalidScheme;
-    fn try_into(self) -> Result<String, Self::Err> {
+impl common::scheme::Type {
+    pub fn try_as_string(&self) -> Result<String, InvalidScheme> {
```
Comment (briansmith) [inline proxy/controller-grpc/src/lib.rs]: This should be `try_to_string`. `as_` is for things that are cheap (free), `to_` is for things that are expensive (e.g. allocating).

**Pull request merged**